dataservicecatalog/lib/mappers.py

```diff
@@ -0,0 +1,174 @@
+from typing import List
+from rdflib import Graph, Literal, BNode, Namespace, RDF, URIRef
+from os import environ as env
+URL = env.get("HOST_URL", "http://localhost:8080")
+PUBLISHER_URL = env.get("PUBLISHER_URL")
+
+
+class Catalog:
+
+    def __init__(self, document):
+        assert document.doc_id is not None, "the document must have an id"
+        self.id = str(document.doc_id)
+        if 'publisher' in document:
+            self.publisherUrl = PUBLISHER_URL + "/" + document['publisher']
+        if 'title' in document:
+            self.title = document['title']
+        if 'description' in document:
+            self.description = document['description']
+        if 'dataservices' in document:
+            self.dataservices = []
+            for d in document['dataservices']:
+                self.dataservices.append(d)
+
+
+class Contact:
+    def __init__(self, document):
+        assert document['name'] is not None, "the contact must have a name"
+        self.name = document['name']
+        if 'email' in document:
+            self.email = document['email']
+        if 'url' in document:
+            self.url = document['url']
+
+
+class DataService:
+
+    def __init__(self, document):
+        assert document.doc_id is not None, "the document must have an id"
+        self.id = str(document.doc_id)
+        if 'endpointdescription' in document:
+            self.endpointdescription = document['endpointdescription']
+        if 'title' in document:
+            self.title = document['title']
+        if 'description' in document:
+            self.description = document['description']
+        if 'endpointUrl' in document:
+            self.endpointUrl = document['endpointUrl']
+        if 'contact' in document:
+            self.contactpoint = Contact(document['contact'])
+
+
+def _add_catalog_to_graph(g: Graph, catalog: Catalog) -> Graph:
+    """Adds the catalog to the Graph g and returns g"""
+    dct = Namespace('http://purl.org/dc/terms/')
+    g.bind('dct', dct)
+    dcat = Namespace('http://www.w3.org/ns/dcat#')
+    g.bind('dcat', dcat)
+
+    uri = URL + "/catalogs/" + catalog.id
+
+    # Add triples using store's add method.
+    g.add((URIRef(uri), RDF.type, dcat.Catalog))
+    g.add((URIRef(uri), dct.publisher, URIRef(catalog.publisherUrl)))
+    g.add((URIRef(uri), dct.title, Literal(catalog.title, lang='nb')))
+
+    return g
+
+
+def _add_dataservice_to_graph(g: Graph, dataservice: DataService) -> Graph:
+    """Adds the dataservice to the Graph g and returns g"""
+    dct = Namespace('http://purl.org/dc/terms/')
```

Comment:
Hadde sikkert vært mer oversiktelig med definisjonen av alle namespaces i en egen fil, i stedet for hvor de tilfeldigvis blir brukt

Reply:
Enig. Sleit litt med den. Har oppretta issue #4